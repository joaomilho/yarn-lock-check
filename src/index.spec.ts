import { yarnLockCheck } from "./";
import path from "path";

function useFixture(fixture: string) {
  process.chdir(path.resolve(__dirname, "fixtures", fixture));
}

describe("yarnLockCheck", () => {
  it("does not throw if all registries match", () => {
    useFixture("ok");

    expect(() => yarnLockCheck("https://registry.com")).not.toThrow();
  });

  it("does not throw if packages have no resolve field", () => {
    useFixture("internal");

    expect(() => yarnLockCheck("https://registry.com")).not.toThrow();
  });

  it("throws if any yaml.locks contains wrong registries", () => {
    useFixture("error");

    expect(() =>
      yarnLockCheck("https://registry.com")
    ).toThrowErrorMatchingSnapshot();
  });
});
