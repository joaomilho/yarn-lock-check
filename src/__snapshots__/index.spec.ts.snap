// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yarnLockCheck throws if any yaml.locks contains wrong registries 1`] = `
"
You pushed changes to the yarn.lock with the wrong registry.
Please ensure your registry is set to https://registry.com:

packages/package-error/yarn.lock
    https://wrong-registry.com/test-lib

yarn.lock
    https://wrong-registry.com/test-lib

"
`;
